File: das_backend/src/main/java/ch/sbb/backend/logging/application/TenantContext.java
package ch.sbb.backend.logging.application;

import ch.sbb.backend.logging.domain.model.TenantId;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.oauth2.jwt.Jwt;

public class TenantContext {

    private final TenantId tenantId;

    private TenantContext() {
        Jwt jwt = (Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        String tenantIdString = (String) jwt.getClaims().get("tid");
        tenantId = new TenantId(tenantIdString);
    }

    public static TenantContext current() {
        return new TenantContext();
    }

    public TenantId getTenantId() {
        return tenantId;
    }
}

File: das_backend/src/main/kotlin/ch/sbb/backend/logging/application/TenantContext.kt
package ch.sbb.backend.logging.application

import ch.sbb.backend.logging.domain.TenantId
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.oauth2.jwt.Jwt

class TenantContext private constructor() {
    val tenantId: TenantId

    init {
        val jwt = SecurityContextHolder.getContext().authentication.principal as Jwt
        val tenantIdString = jwt.claims["tid"] as String
        tenantId = TenantId(tenantIdString)
    }

    companion object {
        fun current(): TenantContext {
            return TenantContext()
        }
    }
}
